### ISSUE TITLE
How to perform an inference task with a model provided with trained parameters?

### ISSUE THREAD

**Issue by YMX-zknu** [twke18/HSG#6]
Hi author, I have downloaded the Cityscapes Trained Models and the file is as follows: 
![2](https://github.com/twke18/HSG/assets/39787386/9bfce544-5ecf-483d-9738-5c8e38ddce8c)

I would like to know how to set up the file to perform inference on the Cityspaces dataset with the model you provided.

thanks.

**Comment by twke18** [comment]
Hi,

You can use [these python scripts](https://github.com/twke18/HSG/blob/8de138f7d90ce4f87f29ca274e67f60e48ff09e2/bashscripts/cityscapes/train.sh#L134-L151) for evaluation.  We perform nearest neighbor retrieval to evaluate the trained models.

**Comment by YMX-zknu** [comment]
Thanks, but I'm having another problem, as shown below.
![1711542072519(1)](https://github.com/twke18/HSG/assets/39787386/39eba066-5fa6-4830-ab0a-75d03ca4ca14)

I tried to run the file "generate_pmi_owt_ucm.m" but I did not find the files corresponding to variables 'matDir', 'segDir', 'imgNames' in the program files you provided.

**Comment by twke18** [comment]
Sorry that I could not give you exact examples, as I lost the access to my code repo.  I will instead explain what these variables are:

1. `matDir` is a cache directory which saves the edge maps computed by PMI.  The code will create the directory.
2. `segDir` is the directory which saves all the segmentation masks by thresholding the edge maps.  The code will also create the directory.
3. `imgNames` is a list of image file names which can be found under `rootDir` ([code](https://github.com/twke18/HSG/blob/8de138f7d90ce4f87f29ca274e67f60e48ff09e2/tools/generate_pmi_owt_ucm.m#L35C24-L35C31)).  Originally, `imgNames` are read from these [lines](https://github.com/twke18/HSG/blob/8de138f7d90ce4f87f29ca274e67f60e48ff09e2/tools/generate_pmi_owt_ucm.m#L16-L17).  You can modify it in your way to read the image names accordingly.

**Comment by YMX-zknu** [comment]
Thank you, there are some errors when executing the `inferance.py` file, I hope you can answer them, as described below:

I have constructed the Cityscapes dataset as per your requirement as shown below:
![image](https://github.com/twke18/HSG/assets/39787386/6d55dab3-d53c-45d8-9f8e-c92e69a7f836)

However, while executing the `inferance.py`  file, the following error occurs:
![image](https://github.com/twke18/HSG/assets/39787386/d54621a3-c2f5-42e9-9d96-a1806c596738)

The displayed "gtFineInstId" seems to require instance segmentation images, but this file is not present in the built Cityscapes dataset. What is the reason for this? Thank you very much!

**Comment by twke18** [comment]
Hi, the files are the ground-truth instance masks of Cityscapes.  They are not used for inference at all.  You can either replace them with some dummy files, or download from [here](https://github.com/twke18/HSG/blob/master/misc/gtFineInstId.zip).

**Comment by YMX-zknu** [comment]
Thank you very much, I downloaded the [gtFineInstId.zip](https://github.com/twke18/HSG/blob/master/misc/gtFineInstId.zip) file you provided and am currently able to run the entire program.

I would like to train a new model based on my dataset to replace the pre-trained model you provided. Currently encountering two problems:
1) As shown below, the variable `imgNames` in the [generate_pmi_owt_ucm.m](https://github.com/twke18/HSG/blob/master/tools/generate_pmi_owt_ucm.m) file reads from the `cityscapes_train.txt` file. Is the `cityscapes_train.txt` file the path to the images within all the sub-files of the `leftImg8bit/train` file in the established Cityscapes dataset?
![image](https://github.com/twke18/HSG/assets/39787386/e583850d-26fd-4820-b848-59d5e5d709ee)
Also, what is the purpose of these '*.txt' files (shown below) under the `datasets/cityscapes` path in your project?
![image](https://github.com/twke18/HSG/assets/39787386/ddf33805-401b-4a9c-b8ae-32768a912c93)

2) I want to modify my dataset to the layout of the Cityscapes dataset you provided, but I don't know how the contents of the folder `gtFineId` are generated? Is it generated based on the `gtFine_trainvaltest.zip` file (shown below) on the Cityscapes website? Is it convenient for you to provide the corresponding processing code?
![image](https://github.com/twke18/HSG/assets/39787386/f21773d7-5a0e-4d76-a0d2-a5cec6d5c717)

Thank you very much for your time and help and I look forward to hearing from you.
